fastify.js

```diff
@@ -335,6 +338,43 @@ function fastify (options) {
     }
   }
 
+  function ready (cb) {
+    let resolveReady
+    let rejectReady
+
+    setImmediate(runHooks)
```

Comment:
Why is this needed? It looks like a code smell.

Reply:
just to return the promise first and then run the hooks, it should not be necessary

Reply:
Then it should be a `nextTick`, and also add a comment.

Reply:
I was thinking `nextTick` as well.

This section really helps when making a decision between the two -- https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#process-nexttick

Reply:
Done, my bad: wanted to use nexttick but I wrote immediate 🤦🏻‍♂️

This has solved the `lolex` problem too